tokio/src/runtime/task/core.rs

```diff
@@ -202,6 +222,7 @@ impl<T: Future, S: Schedule> Core<T, S> {
     pub(super) fn drop_future_or_output(&self) {
         // Safety: the caller ensures mutual exclusion to the field.
         unsafe {
+            let _guard = TaskIdGuard::enter(self.task_id);
             self.set_stage(Stage::Consumed);
         }
```

Comment:
Maybe this should happen inside `set_stage`?  Currently you're missing it in `store_output`.

Reply:
I thought of it, but I think -- but not sure -- that a call to `store_output` is always preceded by a call to `drop_future_or_output`, which already destructs the future in the stage, so the guards are unnecessary. Anyway, moving it back to `set_stage`.